### PR TITLE
available to post comments

### PR DESCRIPTION
**Pull request by MolchanovaA**

### CODE REVIEW

File: src/components/Comments.jsx [hunk 1]
```diff
@@ -2,33 +2,58 @@ import { useEffect, useState } from "react";
...
   }, []);
+
+  useEffect(() => {
+    setCommentsToArticle((currentComments) => {
+      return [addComment, ...currentComments];
+    });
+  }, [addComment]);
+
+  useEffect(() => {
+    console.log("from 2nd use E", commentsToArticle);
```
Comment (ghost) [inline src/components/Comments.jsx]: Please remove console logs from code reviews.

File: src/components/Form.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      username: userName,
+      body: commentBody,
+    };
+
+    if (userName && commentBody) {
+      postComment(article_id, postBody).then(({ data: { posted_comment } }) => {
+        setAddComment(posted_comment);
+        setCommentBody("");
+      });
+      alert("posting in progress");
```
Comment (ghost) [inline src/components/Form.jsx]: I would recommend a different way to tell the user that it is working. As you are optimistically rendering the comments, the instant feedback that the comment is on the page is more than enough. Alerts take your user away from the page and can be very disruptive to their user experience, they're also terrible on mobile.

File: src/components/Form.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    } else {
+      alert("add comment");
```
Comment (ghost) [inline src/components/Form.jsx]: As above, is there a better way you could inform the user that they cannot post an empty comment? Even a 'required' on the form is good for this.

File: src/components/Form.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  const commentBodyHandler = (e) => {
+    setCommentBody(e.target.value);
+  };
+
+  return (
+    <form method="POST" className="new-comment" onSubmit={newCommentHandler}>
+      <label>
+        {" "}
+        {userName} Says:
+        <textarea
```
Comment (ghost) [inline src/components/Form.jsx]: Text area's are great for comments. Have a look at a couple of things with it,
1 - currently I can grab and resize the area and move it all around, this _might_ mess with your ui (move your mouse to the bottom right hand corner of it to see what I mean)
2 - Yay for multi line input, but I can't submit the form with keyboard alone, tab does not move me to the add button, I have to manually click it.

**Review by ghost — CHANGES_REQUESTED** [review]
Very close on this one, some comments to consider. But the main point is that if I am offline there is no feedback to the user that there is a problem. The alert comes up telling me that it is being posted, I click ok and my comment is still there. Something that tells me there is an issue would be ideal. Have a look and re submit when you're ready.

**Review by nc-poppy-m — APPROVED** [review]
Great work! My only suggestion to you is that you don't hide the comment box upon submitting. If I want to submit multiple comments, this would be difficult. Instead of changing the form to a confirmation or error message consider making it an element that exists alongside the comment box (e.g. is only visible if there's a message). This would be a much better UI. However, this is up to you. I'll approve this either way

**Pull request merged**